### ISSUE TITLE
Wrong libtiff referenced while installing `r-sf` on macOS Ventura 13.4.1

### ISSUE THREAD

**Issue by ptr-64** [r-spatial/sf#2228]
I followed the [install instructions on r-sf](https://r-spatial.github.io/sf/#installing) to the letter but when I pass `install.packages("sf", type = "source", configure.args = "--with-proj-lib=$(brew --prefix)/lib/")`, installation fails because `/opt/homebrew/Cellar/libgeotiff/1.7.1_2/lib/libgeotiff.5.dylib` references `/opt/homebrew/opt/libtiff/lib/libtiff.5.dylib`, and I have `/opt/homebrew/opt/libtiff/lib/libtiff.6.dylib` installed.

None of the other issues on github seemed to help and I can't find how to install `libtiff.5.dylib`. Full error message posted below:

```
checking GDAL: checking whether PROJ is available for running:... dyld[57184]: Library not loaded: /opt/homebrew/opt/libtiff/lib/libtiff.5.dylib
  Referenced from: <0CAB49EB-EF79-36B6-A94D-C6319A10CEF5> /opt/homebrew/Cellar/libgeotiff/1.7.1_2/lib/libgeotiff.5.dylib
  Reason: tried: '/opt/homebrew/opt/libtiff/lib/libtiff.5.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/opt/libtiff/lib/libtiff.5.dylib' (no such file), '/opt/homebrew/opt/libtiff/lib/libtiff.5.dylib' (no such file), '/usr/local/lib/libtiff.5.dylib' (no such file), '/usr/lib/libtiff.5.dylib' (no such file, not in dyld cache), '/opt/homebrew/Cellar/libtiff/4.5.1/lib/libtiff.5.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/Cellar/libtiff/4.5.1/lib/libtiff.5.dylib' (no such file), '/opt/homebrew/Cellar/libtiff/4.5.1/lib/libtiff.5.dylib' (no such file), '/usr/local/lib/libtiff.5.dylib' (no such file), '/usr/lib/libtiff.5.dylib' (no such file, not in dyld cache)Library not loaded: /opt/homebrew/opt/hdf5/lib/libhdf5_hl.200.dylib
  Referenced from: <43F452C4-D7EF-334E-8698-DBD4E6870EEE> /opt/homebrew/Cellar/netcdf/4.9.2/lib/libnetcdf.19.dylib
  Reason: tried: '/opt/homebrew/opt/hdf5/lib/libhdf5_hl.200.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/opt/hdf5/lib/libhdf5_hl.200.dylib' (no such file), '/opt/homebrew/opt/hdf5/lib/libhdf5_hl.200.dylib' (no such file), '/usr/local/lib/libhdf5_hl.200.dylib' (no such file), '/usr/lib/libhdf5_hl.200.dylib' (no such file, not in dyld cache), '/opt/homebrew/Cellar/hdf5/1.14.2/lib/libhdf5_hl.200.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/Cellar/hdf5/1.14.2/lib/libhdf5_hl.200.dylib' (no such file), '/opt/homebrew/Cellar/hdf5/1.14.2/lib/libhdf5_hl.200.dylib' (no such file), '/usr/local/lib/libhdf5_hl.200.dylib' (no such file), '/usr/lib/libhdf5_hl.200.dylib' (no such file, not in dyld cache)
./configure: line 3665: 57184 Abort trap: 6           ./gdal_proj
no
configure: error: OGRCoordinateTransformation() does not return a coord.trans: PROJ not available?
ERROR: configuration failed for package ‘sf’
* removing ‘/opt/homebrew/lib/R/4.3/site-library/sf’

The downloaded source packages are in
	‘/private/var/folders/ly/8rr2zmdd2b715s9h80vxjz_80000gn/T/Rtmpb0V7ns/downloaded_packages’
Warning message:
In install.packages("sf", type = "source", configure.args = "--with-proj-lib=/opt/homebrew/lib/") :
  installation of package ‘sf’ had non-zero exit status
```

**Comment by edzer** [comment]
Is there a reason why you don't do a binary install from CRAN or r-universe?

**Comment by ptr-64** [comment]
> Is there a reason why you don't do a binary install from CRAN or r-universe?

Ok I tried to install binaries from r-universe all day yesterday and the package wouldn't load. I spent all of today trying to get it to compile from source (also failed hence my question).
I just tried installing the binary but downloaded the tgz file from CRAN and it worked (I have no idea why).  
Closing the question, thank you for suggesting CRAN binaries (I thought the binaries would be exactly the same between r-universe and CRAN)
